Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -1305,6 +1212,218 @@ def get_exclusion(self, limit, tenant_id=None, filter=None):
+                'operator': 'in',
+                'value': [isolate]
+            })
+
+        if hostname:
+            filters.append({
+                'field': 'hostname',
+                'operator': 'in',
+                'value': hostname
+            })
+
+        if first_seen_gte:
+            filters.append({
+                'field': 'first_seen',
+                'operator': 'gte',
+                'value': first_seen_gte
+            })
+
+        if first_seen_lte:
+            filters.append({
+                'field': 'first_seen',
+                'operator': 'lte',
+                'value': first_seen_lte
+            })
+
+        if last_seen_gte:
+            filters.append({
+                'field': 'last_seen',
+                'operator': 'gte',
+                'value': last_seen_gte
+            })
+
+        if last_seen_lte:
+            filters.append({
+                'field': 'last_seen',
+                'operator': 'lte',
+                'value': last_seen_lte
+            })
+
+        return filters
+
+    def add_endpoint_tag(
+        self,
+        endpoint_ids,
+        tag,
+        endpoint_id_list=None,
+        dist_name=None,
+        ip_list=None,
+        group_name=None,
+        platform=None,
+        alias_name=None,
+        isolate=None,
+        hostname=None,
+        first_seen_gte=None,
+        first_seen_lte=None,
+        last_seen_gte=None,
+        last_seen_lte=None,
+        status=None,
+        no_filter=False
+    ):
+
+        if no_filter:
+            filters = self.create_request_filters(endpoint_id_list=endpoint_ids)
+        else:
+            filters = self.create_request_filters(
+                status=status, endpoint_id_list=endpoint_id_list, dist_name=dist_name,
+                ip_list=ip_list, group_name=group_name, platform=platform, alias_name=alias_name, isolate=isolate,
+                hostname=hostname, first_seen_gte=first_seen_gte, first_seen_lte=first_seen_lte,
+                last_seen_gte=last_seen_gte, last_seen_lte=last_seen_lte
+            )
+
+        body_request = {
+            'context': {
+                'lcaas_id': endpoint_ids,
+            },
+            'request_data': {
+                'filters': filters,
+                'tag': tag
+            },
+        }
+
+        return self._http_request(
+            method='POST',
+            url_suffix='/tags/agents/assign/',
+            json_data=body_request,
+            timeout=self.timeout
+        )
+
+    def remove_endpoint_tag(
+        self,
+        endpoint_ids,
+        tag,
+        endpoint_id_list=None,
+        dist_name=None,
+        ip_list=None,
+        group_name=None,
+        platform=None,
+        alias_name=None,
+        isolate=None,
+        hostname=None,
+        first_seen_gte=None,
+        first_seen_lte=None,
+        last_seen_gte=None,
+        last_seen_lte=None,
+        status=None,
+        no_filter=False
+    ):
+        if no_filter:
```

Comment:
kind of a strange method signature to have so many optional params, and a boolean with whether those params have value. Lets call create_request_filters outside and then pass filters into the param here

Reply:
I agree, made some changes in the structure of the code to have more shared code.